e2e_test/batch/types/array_ty.slt.part

```diff
@@ -105,3 +105,123 @@ drop table t;
 # This is different from PostgreSQL, we may want to change this in the future.
 statement ok
 select array[array[1,2], array[3]];
+
+# array range access
+query T
+select array[1,NULL,2][-1:134124523];
+----
+{1,NULL,2}
+
+query T
+select (array[1,NULL,2])[3:1];
+----
+NULL
+
+query T
+select array[array[1],array[2],array[3]][-21432315:134124523];
+----
+{{1},{2},{3}}
+
+query T
+select array[array[1],array[2],array[3]][-21432315:134124523][1];
```

Comment:
just found `select array[array[1],array[2],array[3]][-21432315:134124523][1];` is also different
PG:
```
select (array[array[1],array[2],array[3]])[-21432315:134124523][1];
     array     
---------------
 {{1},{2},{3}}
(1 row)
```